code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/dao/AuthDao.java

```diff
@@ -18,7 +22,29 @@
 
     Account createAccount(String name, String kind, String externalId, String externalType);
 
+    Account createProject(String name, String kind);
+
+    Account createDefaultProject(Account account);
+
     void updateAccount(Account account, String name, String kind, String externalId, String externalType);
 
-    List<? extends Account> getAccessibleProjects(String userAccountId, List<String> orgIds, List<String> teamIds);
+    List<Account> getAccessibleProjects(Set<ExternalId> externalIdSet, Boolean isAdmin, Long policyAccount);
```

Comment:
:) same as the comment below. Just pass policy to the Dao and extract all the needed info there.